lib/preview/routes/module.js

```diff
@@ -0,0 +1,73 @@
+const http = require('../../../http')
+const { fetchModulesByPath } = require('../../../api/designManager');
+const {
+  addRefreshScript,
+  getPreviewUrl,
+  memoize,
+  trackPreviewEvent,
+  buildHTMLResponse
+} = require('../previewUtils');
+const { logger } = require('./../../../logger');
+
+const cachedFetchModulesByPath = memoize(fetchModulesByPath);
+
+const buildModuleRouteHandler = (sessionInfo) => {
+  const { accountId, sessionToken } = sessionInfo;
+
+  return async (req, res) => {
+    trackPreviewEvent('view-module-route');
+
+    const { modulePath } = req.params;
+
+    if (!modulePath) {
+      res.status(200).set({ 'Content-Type': 'text/html' }).end(buildModuleIndex());
+      return;
+    }
+
+    const calculatedPath = `@preview/${sessionToken}/${modulePath}.module`;
+    let customWidgetInfo;
+    try {
+      customWidgetInfo = await cachedFetchModulesByPath(accountId, calculatedPath);
+    } catch (err) {
+      logger.error(`Failed to fetch module preview for ${calculatedPath}`)
+    }
+    if (!customWidgetInfo || !('moduleId' in customWidgetInfo && 'previewKey' in customWidgetInfo)) {
```

Comment:
This seems very brittle--do we have plans to refactor this in the future?

Reply:
Can you explain what you mean by brittle here? I may be too frog-in-boiling-water to see anything iffy 👀